### DISCUSSION TITLE
Support polars dataframes in data viewer

### DISCUSSION THREAD

**Discussion by hagsted** [microsoft/vscode-jupyter#11049] [New Ideas]
I am trying out polars dataframes as a replacement for pandas, so far I am happy with the execution speed gains and syntax. My only complain is that in VS-Code when using Jupyter, I am not able to browse the dataframe in the data viewer, unless I do a .to_pandas(). I would like native support for showing polars dataframes in the data viewer. I can print it out in jupyter, but that is only good for a quick check of the validity.

Best regards Kristian
![image](https://user-images.githubusercontent.com/4332517/182775297-a8d0c6d1-a790-4d91-95af-edec65cd9007.png)

**Comment by mol1337** [comment]
I second this idea. 
When doing data analysis I really like the data viewer for pandas dataframes with its filter functions and nice overview. However, i also experience the same issue with polars dataframes.
Best regards Mathias

**Reply by shiosai** [reply]
#12519

**Comment by NicoleSiawAER** [comment]
100% support this idea. It would be very helpful for a full migration to a polar based workflow
